### DISCUSSION TITLE
[A-L] (3 Aprile 2023) Esercizio 5

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#384] [Esami A-L]
A una festa vengono invitati Marco, Carla e Toni, che rilasciano le seguenti dichiarazioni.

- Marco dice: "O vengo io e non viene Toni oppure veniamo io e Carla."

- Carla dice: "Non vengono nè Marco nè Toni."

- Toni dice: “Marco viene e Carla non viene.”

---

1. Formalizzare le tre affermazioni in logica proposizionale scegliendo un linguaggio adeguato.

2. E possibile che tutti e tre dicano il vero? Argomentare

3. Se tutti dicono il falso, chi parteciperà sicuramente alla festa? Argomentare.

(NB: le argomentazioni vanno svolte preferibilmente in base alla formalizzazione proposta al punto 1 e
utilizzando le nozioni formali di assegnamento, soddisfazione, verità logica, tavola di verità etc.).

**Comment by DavideGalilei** [comment]
### 1. Formalizzazione
M = Marco viene (quando M=1)
C = Carla viene (quando C=1)
T = Toni viene (quando T=1)

- Marco dice: $\large{(M \land \lnot{T}) \lor (M \land C)}$
- Carla dice: $\large{\lnot{M} \land \lnot{T}}$
- Toni dice: $\large{M \land \lnot{C}}$

<img src="https://github.com/user-attachments/assets/55882c42-8adb-4eb8-b115-1c0a8c4b1da9" width="250">

### 2. È possibile che tutti e tre dicano il vero?

Tavola di verità:
| M | C | T | Marco dice (Blu) | Carla dice (Verde) | Toni dice (Giallo) |
|---|---|---|-----------------|-------------------|--------------------|
| 0 | 0 | 0 | 0               | 1                 | 0                  |
| 0 | 0 | 1 | 0               | 0                 | 0                  |
| 0 | 1 | 0 | 0               | 1                 | 0                  |
| 0 | 1 | 1 | 0               | 0                 | 0                  |
| 1 | 0 | 0 | 1               | 0                 | 1                  |
| 1 | 0 | 1 | 0               | 0                 | 1                  |
| 1 | 1 | 0 | 1               | 0                 | 0                  |
| 1 | 1 | 1 | 1               | 0                 | 0                  |

<img src="https://github.com/user-attachments/assets/7d5ceae8-7d3c-482c-864d-95228f404a7a" width="250">

La risposta è **falso**: non esiste nessun ed un solo caso dove le informazioni sono coerenti tra di loro. Per essere vero, avremmo dovuto trovare una sola riga dove le affermazioni hanno valori $\large{(1, 1, 1)}$, ma non è presente.

### 3. Tutti dicono il falso
Se tutti dicono il falso, allora è possibile invertire le loro affermazioni, pertanto la tavolà di verità andrà interpretata comparando i casi dove tutte le tre affermazioni hanno valore $\large{(0, 0, 0)}$.

<img src="https://github.com/user-attachments/assets/1a8810b9-ad0e-4296-9228-39e1b5b3c689" width="250">

Gli unici due casi coerenti sono dove $M,C,T$ sono rispettivamente $\large{(0, 0, 1)}$ e $\large{(0, 1, 1)}$, quindi possiamo dire con certezza solo che $\large{M=0}$ e $\large{T=1}$, ma non si può dire con certezza il valore di $\large{C}$.
